toolkit/tools/pkg/isomakerlib/isomaker.go

```diff
@@ -245,102 +338,150 @@ func (im *IsoMaker) copyShimFromInitrd(efiBootImgTempMountDir, bootBootloaderFil
 	const bootKernelFile = "boot/vmlinuz"
 
 	vmlinuzFilePath := filepath.Join(im.buildDirPath, "isolinux/vmlinuz")
 
 	// In order to select the correct kernel for isolinux, open the initrd archive
 	// and extract the vmlinuz file in it. An initrd is a gzip of a cpio archive.
 	//
-	im.extractFromInitrdAndCopy(bootKernelFile, vmlinuzFilePath)
+	return im.extractFromInitrdAndCopy(bootKernelFile, vmlinuzFilePath)
 }
 
 // createIsoRpmsRepo initializes the RPMs repo on the ISO image
 // later accessed by the ISO installer.
-func (im *IsoMaker) createIsoRpmsRepo() {
+func (im *IsoMaker) createIsoRpmsRepo() (err error) {
 	isoRpmsRepoDirPath := filepath.Join(im.buildDirPath, "RPMS")
 
 	logger.Log.Debugf("Creating ISO RPMs repo under '%s'.", isoRpmsRepoDirPath)
 
-	logger.PanicOnError(os.MkdirAll(isoRpmsRepoDirPath, os.ModePerm), "Failed to mkdir '%s'.", isoRpmsRepoDirPath)
+	err = os.MkdirAll(isoRpmsRepoDirPath, os.ModePerm)
+	if err != nil {
+		return fmt.Errorf("failed to mkdir '%s'.", isoRpmsRepoDirPath)
+	}
 
 	fetchedRepoDirContentsPath := filepath.Join(im.fetchedRepoDirPath, "*")
-	recursiveCopyDereferencingLinks(fetchedRepoDirContentsPath, isoRpmsRepoDirPath)
+	err = recursiveCopyDereferencingLinks(fetchedRepoDirContentsPath, isoRpmsRepoDirPath)
+	if err != nil {
+		return err
+	}
+
+	return nil
 }
 
 // prepareWorkDirectory makes sure we start with a clean directory
 // under "im.buildDirPath". The work directory will contain the contents of the ISO image.
-func (im *IsoMaker) prepareWorkDirectory() {
+func (im *IsoMaker) prepareWorkDirectory() (err error) {
 	logger.Log.Infof("Building ISO under '%s'.", im.buildDirPath)
 
 	exists, err := file.DirExists(im.buildDirPath)
-	logger.PanicOnError(err, "Failed while checking if directory '%s' exists.", im.buildDirPath)
+	if err != nil {
+		return fmt.Errorf("failed while checking if directory '%s' exists:\n%w", im.buildDirPath, err)
+	}
 	if exists {
 		logger.Log.Warningf("Unexpected: temporary ISO build path '%s' exists. Removing.", im.buildDirPath)
-		logger.PanicOnError(os.RemoveAll(im.buildDirPath), "Failed while removing directory '%s'.", im.buildDirPath)
+		err = os.RemoveAll(im.buildDirPath)
+		if err != nil {
+			return fmt.Errorf("failed while removing directory '%s':\n%w", im.buildDirPath, err)
+		}
 	}
 
-	logger.PanicOnError(os.Mkdir(im.buildDirPath, os.ModePerm), "Failed while creating directory '%s'.", im.buildDirPath)
+	err = os.Mkdir(im.buildDirPath, os.ModePerm)
+	if err != nil {
+		return fmt.Errorf("failed while creating directory '%s':\n%w", im.buildDirPath, err)
+	}
 
-	im.deferIsoMakerCleanUp(func() {
+	im.deferIsoMakerCleanUp(func() error {
 		logger.Log.Debugf("Removing '%s'.", im.buildDirPath)
-
-		logger.PanicOnError(os.RemoveAll(im.buildDirPath), "Failed to remove '%s'.", im.buildDirPath)
+		err := os.RemoveAll(im.buildDirPath)
+		if err != nil {
+			err = fmt.Errorf("failed to remove '%s':\n%w", im.buildDirPath, err)
+		}
+		return err
```

Comment:
The code is correct but to avoid confusion with the `err` variable from `prepareWorkDirectory` we tend to give distinct names to the error variables in anonymous functions:
```suggestion
		err := os.RemoveAll(im.buildDirPath)
		if cleanupErr != nil {
			cleanupErr = fmt.Errorf("failed to remove '%s':\n%w", im.buildDirPath, cleanupErr)
		}
		return cleanupErr
```

Reply:
good point - fixed.